### PR TITLE
Support bevy v0.13; Fix multiple input

### PR DESCRIPTION
**Pull request by shanecelis**
The first commit will make `bevy-input-sequence` work with bevy v0.13. That first commit has no provocative changes whatsoever unlike the following commits.

The commits following that try to deal with the bigger issues of `ActPattern` not working properly. I ended up removing `Act` and `ActPattern` altogether. I think it's come out simpler and has the same functionality.

I then found an issue with the patch I submitted to `trie-rs` where if you added key seq "W D S A" then added "W D", it looked like only "W D S A" was added. But if you added them "W D" first then it would work. Here's the [PR for trie-rs](https://github.com/laysakura/trie-rs/pull/21).

I also added a `GamepadEvent` trait that allows us to report what gamepad matched a sequence. That trait is only required on events when used with `ButtonSequence`, which is my name for gamepad button sequences. On the plus side, this makes it clear that we don't support mixed keyboard and gamepad sequences. I left the remnant of the mixed input test with a comment. We can resurrect this feature if you think it's important.

Finally I bumped the version to 0.3 and marked it as compatible with bevy 0.13 in the README.

Take a look and let me know what you think. Thanks!

-Shane

### CODE REVIEW

**Comment by not-elm** [comment]
Thounk you for PR.

I don't think there is any problem with deleting `ActPattern::Any`.

I tried to build library in my environment but it failed, so could you please fix dependecies for `keyseq` in `Cargo.toml`?

Also, when I saw GamepadEvent, I came up with a new idea. )
> [!NOTE]
> Please rest assured that these ideas are for the future and  are not intended to be incorporated into this PR.

Currently we are passing the event itself when spawning the sequence, but we might also be able to pass a callback that creates the event separately; That would allow us to create the 
events without using GamepadEvent as bellow.

```rust
   commands.spawn(ButtonSequence::new(
        [
            GamepadButtonType::North,
            GamepadButtonType::East,
            GamepadButtonType::South,
            GamepadButtonType::West,
        ],
        |gamepad: Gamepad|{
            MyEvent(gamepad)
        }
    ).time_limit(Duration::from_secs(1)));
```
Also, I think it would be interesting to be able to link to [`bevy_mod_picking`].(https://docs.rs/bevy_mod_picking/latest/bevy_mod_picking/).

**Comment by shanecelis** [comment]
Great! Removed path from keyseq dependency. Try the build again.

I like the idea of using a closure, then we don't even have to require that the event be clone. However, we'd need to know that closure at app setup-time rather than when the sequence is defined because `InputSequence<E,K>` would become `InputSequence<E,K,F>` otherwise. I'll mull it over.

File: tests/act.rs [hunk 1]
```diff
@@ -1,119 +1,143 @@
...
+    assert_eq!((Modifiers::CONTROL, KeyCode::Enter), key! { ctrl-Enter });
+    assert_eq!((Modifiers::CONTROL, KeyCode::Space), key! { ctrl-Space });
+    assert_eq!((Modifiers::CONTROL, KeyCode::Tab), key! { ctrl-Tab });
+    assert_eq!((Modifiers::CONTROL, KeyCode::Delete), key! { ctrl-Delete });
     assert_eq!((Modifiers::CONTROL, KeyCode::Minus), key! { ctrl-- });
-    assert_eq!((Modifiers::CONTROL, KeyCode::Underline), key! { ctrl-_ });
-    assert_eq!((Modifiers::CONTROL, KeyCode::Colon), key! { ctrl-: });
     assert_eq!(
-        (Modifiers::CONTROL, KeyCode::Apostrophe),
-        key! { ctrl-'\'' }
```
Comment (not-elm) [inline tests/act.rs:80]: `cargo test` was failed; see below for the error log.

Once corrected, i would like to merge and prepare for release!

```
error[E0599]: no variant or associated item named `Grave` found for enum `bevy::prelude::KeyCode` in the current scope
   --> tests\act.rs:125:58
    |
125 |     assert_eq!((Modifiers::CONTROL, KeyCode::Backquote), key! { ctrl-'`' });
    |                                                          ^^^^^^^^^^^^^^^^^ variant or associated item not found in `KeyCode` 
```

Comment (shanecelis) [reply tests/act.rs:80]: Doh! There are commits to keyseq on my desktop computer that I need to push and publish. Unfortunately I'm going to be away from my desktop computer until Thursday. :( Apologies. I'll comment here when I get that sorted out.

Comment (not-elm) [reply tests/act.rs:80]: Yeah, Best regards!

**Comment by shanecelis** [comment]
Thanks for your patience. New version of `keyseq` is pushed and published. Updated version for `bevy-input-sequence`. It should work now.

**Comment by not-elm** [comment]
I'm sorry I'm late.

There seems to be no problem with the pull request, so I will merge it.
Thank you!

**Pull request merged**